src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -158,26 +158,34 @@ public override async Task RunAsync(AgentTaskPluginExecutionContext executionCon
             executionContext.Debug($"Repository requires to be placed at '{expectRepoPath}', current location is '{currentRepoPath}'");
             if (!string.Equals(currentRepoPath.Trim(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar), expectRepoPath.Trim(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar), IOUtil.FilePathStringComparison))
             {
-                executionContext.Output($"Repository is current at '{currentRepoPath}', move to '{expectRepoPath}'.");
-                var count = 1;
-                var staging = Path.Combine(tempDirectory, $"_{count}");
-                while (Directory.Exists(staging))
+                // In a multi-checkout scenario we don't move directories, since the previous directory could be another clone in use
```

Comment:
My only change is 161-165.  The diff here isn't great.  Basically if in multi-checkout give a different message and skip the move (which fails / is caught anyways).

Reply:
Unfortunately, I don't remember enough of the details here to provide much guidance. The point of the moving is to avoid costly clones if the path was simply changed for some reason. I think it would be worth understanding why it's failing before changing the code. If I remember correctly this worked at some point, so I don't understand why it would be failing now.

Reply:
@jpricket it definitely works for non-multi checkout

In my example

```
 - checkout: self   
        path: path1
      - checkout: self
        path: path2
```

'path1' is the directory it identifies the existing clone to move.  I don't think we want that.  Not the change I have just removes the warning the move was failling anyways.

Also if you have say 
```
 - checkout: self   
        path: path1
      - checkout: repo2
        path: path2
```
You won't attempt to move since the remotes don't match

Reply:
@jpricket , you are giving me ideas though, i'll try a few more thinks.  I miss you :-)!

Reply:
So, this code is about moving between builds. I.e. you start with a build that has 
```
- checkout: self
- checkout: repo2
  path: path1
```
and then in the next build you change it to
```
- checkout: self
- checkout: repo2
  path: path2
```

I don't think we really support checking out the same repo alias to 2 different locations as in your first example.

Reply:
This is great feedback @jpricket !  I had a deeper look and updated with a much better fix.  You are the best!